### PR TITLE
add priority entitlement

### LINKED ISSUE #3771: add Priority Entitlement
<sup>ℹ️ [Figma (root)][figma] | [Sitemap][sitemap]</sup>
<sup>_This initial comment is collaborative and open to modification by all._</sup>

## Description
Add Priority Entitlement information to the app from front to back end

## Screenshot: Prototype
image

## Acceptance Criteria
- [x] add priority entitlement boolean to database, model, and schema (ensure it is added to all the types/inputs needed)
- [x] add priority entitlement string to database, model, and schema
- [x] update factory and fakeUser
- [x] update `/government-info/edit` with the new fields to match screenshot
- [x] update sections of Profile, General Information, Candidate profile with new fields
- [x] update CSV and print profile with new fields

   [image]: https://user-images.githubusercontent.com/305339/166570935-6425a615-82c7-4b2f-9840-5c7d698fa2ab.png
   [image-link]: https://example.com
   [figma]: https://go.talent.c4nada.ca/figma
   [sitemap]: https://go.talent.c4nada.ca/sitemap

### PR DESCRIPTION
**Pull request by vd1992**
closes #3771 
Adds the field from front to back, fullstack addition 

**Testing**: 
adjust field values at `/profile/government-info/edit`, `/profile/create-account`, and with Adminer

ensure things look like they should at
`/profile`
Candidate Profile tab on `/admin/users/:id`
print page
CSV generated

### CODE REVIEW

**Comment by vd1992** [comment]
will update main and then run storybook check shortly

**Review by JamesHuf — APPROVED** [review]
Everything looks good to me! The only thing we might want changed is that users aren't limited to just a number for priority number but can include other characters. This doesn't break anything though so not really an issue.

**Comment by vd1992** [comment]
> Everything looks good to me! The only thing we might want changed is that users aren't limited to just a number for priority number but can include other characters. This doesn't break anything though so not really an issue.

Ah, but priority number I have set to being a string though? It isn't a number at all

File: api/database/factories/UserFactory.php [hunk 1]
```diff
@@ -135,6 +136,8 @@ public function definition()
             'gov_employee_type' => $this->faker->randomElement(ApiEnums::govEmployeeTypes()),
             'citizenship' => $this->faker->randomElement(ApiEnums::citizenshipStatuses()),
             'armed_forces_status' => $this->faker->randomElement(ApiEnums::armedForcesStatuses()),
+            'has_priority_entitlement' => $hasPriorityEntitlement,
+            'priority_number' => $hasPriorityEntitlement? $this->faker->word() : null,
```
Comment (vd1992) [inline api/database/factories/UserFactory.php:140]: @JamesHuf 
faker is generating priority number as a string even

**Pull request merged**